ansible_runner/utils/capacity.py

```diff
@@ -21,17 +26,24 @@ def get_mem_in_bytes():
         return error
 
 
-def get_uuid():
-    try:
-        with open('/var/lib/dbus/machine-id') as f:
-            uuid = f.read()
-        return uuid.strip()
-    except FileNotFoundError:
-        try:
-            with open('/etc/machine-id') as f:
-                uuid = f.read()
-                return uuid.strip()
-        except FileNotFoundError:
-            error = ("Could not find /var/lib/dbus/machine-id or "
-                     "/etc/machine-id files, UUID undiscoverable.")
-            return error
+def generate_or_get_uuid():
```

Comment:
Functions that combine multiple capabilities like this are generally harder to test and maintain. It's better to have simple, single purpose functions. That allows the caller to control the desired behavior.

This function could be changed to call `get_uuid()`, then save it to disk. That way `get_uuid()` would be concerned with uuid generation and this function would be concerned with making sure the file exists on disk. Maybe call rename it to `save_uuid()`, have it accept a path to check or save to as a parameter, and return the value that it either gets from the path or save a new uuid to if the file doesn't exist.